applications/asset_tracker_v2/tests/debug_module/src/debug_module_test.c

```diff
@@ -0,0 +1,236 @@
+	__wrap_module_start_ExpectAndReturn(&expected_module_data, 0);
+
+	TEST_ASSERT_EQUAL(0, DEBUG_MODULE_EVT_HANDLER((struct event_header *)app_module_event));
+
+	k_free(app_module_event);
+}
+
+void test_memfault_trigger_metric_sampling_on_gps_fix(void)
+{
+	setup_debug_module_in_init_state();
+
+	/* Update this function to expect the search time and number of satellites. */
+	__wrap_memfault_metrics_heartbeat_set_unsigned_ExpectAndReturn(
+						MEMFAULT_METRICS_KEY(GpsTimeToFix),
+						60000,
+						0);
+	__wrap_memfault_metrics_heartbeat_set_unsigned_ExpectAndReturn(
+						MEMFAULT_METRICS_KEY(GpsSatellitesTracked),
+						4,
+						0);
+	__wrap_memfault_metrics_heartbeat_debug_trigger_Ignore();
+
+	struct gps_module_event *gps_module_event = new_gps_module_event();
+
+	gps_module_event->type = GPS_EVT_DATA_READY;
+	gps_module_event->data.gps.satellites_tracked = 4;
+	gps_module_event->data.gps.search_time = 60000;
+
+	TEST_ASSERT_EQUAL(0, DEBUG_MODULE_EVT_HANDLER((struct event_header *)gps_module_event));
+
+	k_free(gps_module_event);
+}
+
+void test_memfault_trigger_metric_sampling_on_gps_timeout(void)
+{
+	resetTest();
+	setup_debug_module_in_init_state();
+
+	/* Update this function to expect the search time and number of satellites. */
+	__wrap_memfault_metrics_heartbeat_set_unsigned_ExpectAndReturn(
+						MEMFAULT_METRICS_KEY(GpsTimeoutSearchTime),
+						30000,
+						0);
+	__wrap_memfault_metrics_heartbeat_set_unsigned_ExpectAndReturn(
+						MEMFAULT_METRICS_KEY(GpsSatellitesTracked),
+						2,
+						0);
+	__wrap_memfault_metrics_heartbeat_debug_trigger_Ignore();
+
+	struct gps_module_event *gps_module_event = new_gps_module_event();
+
+	gps_module_event->type = GPS_EVT_TIMEOUT;
+	gps_module_event->data.gps.satellites_tracked = 2;
+	gps_module_event->data.gps.search_time = 30000;
+
+	TEST_ASSERT_EQUAL(0, DEBUG_MODULE_EVT_HANDLER((struct event_header *)gps_module_event));
+
+	k_free(gps_module_event);
+}
+
+void test_memfault_trigger_data_send(void)
+{
+	resetTest();
+	setup_debug_module_in_init_state();
+
+	__wrap__event_submit_ExpectAnyArgs();
+
+	__wrap_memfault_packetizer_data_available_ExpectAndReturn(1);
+	__wrap_memfault_packetizer_get_chunk_ExpectAnyArgsAndReturn(1);
+	__wrap_memfault_packetizer_get_chunk_ExpectAnyArgsAndReturn(0);
+
+	/* Expect the debug module to generate an event with accompanied Memfault metric data. */
+	__wrap__event_submit_Stub(&validate_debug_data_ready_evt);
+
+	struct data_module_event *data_module_event = new_data_module_event();
+
+	data_module_event->type = DATA_EVT_DATA_SEND;
+
+	TEST_ASSERT_EQUAL(0, DEBUG_MODULE_EVT_HANDLER((struct event_header *)data_module_event));
+
+	k_free(data_module_event);
+}
+
+void test_memfault_unhandled_event(void)
+{
+	resetTest();
+	setup_debug_module_in_init_state();
+
+	/* Expect no memfault APIs to be called on GPS_EVT_ACTIVE */
+
+	struct gps_module_event *gps_module_event = new_gps_module_event();
+
+	gps_module_event->type = GPS_EVT_ACTIVE;
```

Comment:
Can you extend this test to send all unhandled events? (If thats cumbersome, please change the name of this test to test_memfault_no_memfault_call_on_gps_evt_active())

Reply:
Jepp. will add test for all other unhandled events.